src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -380,10 +413,16 @@ impl StaticHeader {
                 // Copy 2 read OK, 1 resulted in IO Error
                 match StaticHeader::sigblock_from_buf(&buf_loc_2) {
                     Ok(loc_2) => {
-                        if loc_2.is_some() {
-                            BDA::write(f, &buf_loc_2, MetadataLocation::First)?;
+                        if let Some(loc_2) = loc_2 {
```

Comment:
Note that in this if-expression you always return ```Ok(...)```, so you can move the ```Ok``` out to surround the ```if-expression```, like ```Ok(if let Some(loc_2) = loc_2 {...} else { None })```. It's a bit clearer that way, because it reinforces the fact that if you found one good signature buffer and the other just had an IO error, then you consider the succesfully read one to be correct.